query.go

```diff
@@ -661,9 +692,15 @@ func (s *ChainService) GetCFilter(blockHash chainhash.Hash,
 	getHeader := s.RegFilterHeaders.FetchHeader
 	dbFilterType := filterdb.RegularFilter
 
-	// First check the database to see if we already have this filter. If
+	// First check the cache to see if we already have this filter. If
 	// so, then we can return it an exit early.
-	filter, err := s.FilterDB.FetchFilter(&blockHash, dbFilterType)
+	filter, err := s.getFilterFromCache(&blockHash, dbFilterType)
+	if err == nil && filter != nil {
```

Comment:
We should return if the errors are unexpected. (as in `ErrFilterNotFound` is an expected error from `FetchFilter` below). Same for `FilterBD.FetchFilter`.

Reply:
added this as a separate commit for now, wouldn't failing `GetCFilter` for an error in `FilterDB` or `Cache` (which are not *that* critical) lead to smaller bugs in those 2 systems becoming bigger bugs in neutrino ?

Reply:
Well the thing is that GetCFilter's goal is to get a filter by whatever means, the local db and cache is just a way of making it faster, I don't think any failure in them is critical, sure it would make getting filters slower, but the other option would be not getting anything.

For example, I could imagine iOS killing the app in the middle of a run and corrupting filterdb, that doesn't mean GetCFilter should stop working, maybe we could log it instead ?

Either way, I left the code returning the error if it's anything else than ErrElementNotFound and ErrFilterNotFound, but I think this is less than ideal.